### ISSUE TITLE
API Routes Sometimes Don't Work On New Versions Of SolidStart

### ISSUE THREAD

**Issue by OrJDev** [solidjs/solid-start#377]
So i was having issues on api routes on my trpc project and decided to see wether its on my side or your side, 
i created new project using `npm init solid` yet when im trying to make an api requet it redirects me to index.html

as you can see the api route seems to be in there yet it says index.html everytime i make a request to the api (should be /api/testing)

![Screenshot_1](https://user-images.githubusercontent.com/91349014/198683142-78009079-6729-453b-8e32-0c8dd4d978b4.png)
![Screenshot_2](https://user-images.githubusercontent.com/91349014/198683163-56ae7c89-ac56-4183-a032-be70c8843720.png)

**Comment by OrJDev** [comment]
seems to be working fine on older projects of solidStart like https://github.com/OrJDev/solid-trpc-authentication

**Comment by OrJDev** [comment]
Seems to be working when posting data
![Screenshot_1](https://user-images.githubusercontent.com/91349014/198692643-142a75b0-5e5f-4100-87f0-f6d003a013f5.png)
but get requests are being redirected to index.html
![Screenshot_2](https://user-images.githubusercontent.com/91349014/198692743-ce599fd9-8ef2-4525-b128-19daf31a97a2.png)

**Comment by davidestal** [comment]
Tried out solid-start (bare/ts/no ssr) and run into the same issue.
In my case posting data doesn't seem to work. But that might be me doing something wrong.

Sending no payload.
```
POST /api/students
TypeError: Cannot read properties of undefined (reading 'includes')
```

Sending an empty json object.
```
POST /api/students
TypeError: RequestInit: duplex option is required when sending a body.
```

**Comment by ghalle** [comment]
This is an issue since vite 3.2.0, it should be fixed by https://github.com/solidjs/solid-start/pull/373

**Comment by OrJDev** [comment]
> Tried out solid-start (bare/ts/no ssr) and run into the same issue.
> 
> In my case posting data doesn't seem to work. But that might be me doing something wrong.
> 
> 
> 
> Sending no payload.
> 
> ```
> 
> POST /api/students
> 
> TypeError: Cannot read properties of undefined (reading 'includes')
> 
> ```
> 
> 
> 
> Sending an empty json object.
> 
> ```
> 
> POST /api/students
> 
> TypeError: RequestInit: duplex option is required when sending a body.
> 
> ```

That's an undici error so run 

npm install undici@5.11.0

Fixed the post one for me but not sure how to fix the get one right now

**Comment by OrJDev** [comment]
> This is an issue since vite 3.2.0, it should be fixed by https://github.com/solidjs/solid-start/pull/373

Awesome thanks

**Comment by davidestal** [comment]
> > Tried out solid-start (bare/ts/no ssr) and run into the same issue.
> > In my case posting data doesn't seem to work. But that might be me doing something wrong.
> > Sending no payload.
> > ```
> > 
> > POST /api/students
> > 
> > TypeError: Cannot read properties of undefined (reading 'includes')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Sending an empty json object.
> > ```
> > 
> > POST /api/students
> > 
> > TypeError: RequestInit: duplex option is required when sending a body.
> > ```
> 
> That's an undici error try running
> 
> npm install undici@5.11.0
> 
> To fix the duplex option is required error

Thanks!
Going from 5.12.0 to 5.11.0 worked.

**Comment by OrJDev** [comment]
fixed in ^0.2.1
